src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -381,6 +381,32 @@ private static void AddPackageTargetFallbacks(PackageSpec spec, IEnumerable<IMSB
             }
         }
 
+        /// <summary>
+        /// Remove duplicates and excluded values a set of sources or fallback folders.
+        /// </summary>
+        /// <remarks>Compares with Ordinal, excludes must be exact matches.</remarks>
+        public static IEnumerable<string> AggregateSources(IEnumerable<string> values, IEnumerable<string> excludeValues)
+        {
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            if (excludeValues == null)
+            {
+                throw new ArgumentNullException(nameof(excludeValues));
+            }
+
+            var result = new SortedSet<string>(values, StringComparer.Ordinal);
+
+            foreach (var exclude in excludeValues)
```

Comment:
I think you can simplify the `foreach` using a built-in set operation:

    result.ExceptWith(excludeValues);